balena/models/api_key.py

```diff
@@ -15,235 +20,169 @@ class ApiKey:
     def __init__(self):
         self.application = Application()
         self.auth = Auth()
-        self.base_request = BaseRequest()
         self.device = Device()
-        self.settings = Settings()
-
-    def __get_all_with_filter(self, filters):
-        return self.base_request.request(
-            "api_key",
-            "GET",
-            params=filters,
-            endpoint=self.settings.get("pine_endpoint"),
-        )["d"]
 
-    def create_api_key(self, name, description=None):
+    def create(
+        self,
```

Comment:
> My take was not to use `@staticmethod` on everything to avoid making this PR even huger than it is, but it is for sure something that can be discussed

Indeed, that was my line of thinking too!